bigtable/google/cloud/bigtable/row_data.py

```diff
@@ -444,12 +448,17 @@ def __iter__(self):
                     raise ValueError("The row remains partial / is not committed.")
                 break
 
-            for chunk in response.chunks:
-                self._process_chunk(chunk)
-                if chunk.commit_row:
-                    self.last_scanned_row_key = self._previous_row.row_key
-                    self._counter += 1
-                    yield self._previous_row
+            try:
+                for chunk in response.chunks:
+                    if self._stop:
+                        raise StopIteration
+                    self._process_chunk(chunk)
+                    if chunk.commit_row:
+                        self.last_scanned_row_key = self._previous_row.row_key
+                        self._counter += 1
+                        yield self._previous_row
+            except StopIteration:
+                break
```

Comment:
ISTM that if `self._stop` is True, then the call to `self._read_next_response()` should be skipped as well as the chunk processing.  E.g:

```python
        while not self._stop:
            try:
                response = self._read_next_response()
            except StopIteration:
                if self.state != self.NEW_ROW:
                    raise ValueError("The row remains partial / is not committed.")
                break

            for chunk in response.chunks:
                if self._stop:
                    break
                self._process_chunk(chunk)
                if chunk.commit_row:
                    self.last_scanned_row_key = self._previous_row.row_key
                    self._counter += 1
                    yield self._previous_row
```

Reply:
Seems to be a better way, thank you. Done.